biocypher/_get.py

```diff
@@ -335,6 +390,71 @@ def _update_cache_record(self, resource: Resource):
         with open(self.cache_file, "w") as f:
             json.dump(self.cache_dict, f, default=str)
 
+    def _get_api_request(self, api: APIRequest):
+        """
+        Send a GET request to the provided API endpoint,
+        cache it in JSON format to the specified file path.
+
+        Args:
+            api(APIRequest): an API instance representing the API request
+                   that is being cached.
+
+        Returns:
+            dict: The JSON response data from the API request,or from cache.
+
+
+        """
+        cache_file_path = f"{api.name}.json"
+        path = os.path.join(self.cache_dir, cache_file_path)
+        if not os.path.exists(path):
+            logger.info(f"Sending GET request to API URL: {api.url}.")
+            response = requests.get(url=api.url)
+            if response.status_code != 200:
+                raise response.raise_for_status()
+            response_data = response.json()
+
+        else:
+            with open(path, "r") as file:
+                logger.info(
+                    f"Attempting to load {api.name} API request data from file: {path}."
+                )
+                response_data = json.load(file)["api_request"]
+
+        return response_data
+
+    def _cache_api_request(self, response_data, api: APIRequest):
```

Comment:
Where is this method used, why is it here?

Reply:
Thank you for the feedback.Initially, my idea was to create a separate function for getting and caching API requests respectively...so according to what you mentioned,  from the user's perspective, the download method should not only handle the downloading of resources but also be the caching API requests, correct?

Reply:
Yeah, the `download` method should not care whether it's `Resource` or `APIRequest`. It should be able to accept arbitrary collections (such as a list) of `Resource`s and `APIRequest`s.

You created the download and cache functions as private methods (leading underscore), which usually means that you do not want the user to interact with them, right?